okhttp/src/main/java/okhttp3/internal/connection/RouteDatabase.kt

```diff
@@ -24,10 +25,19 @@ import okhttp3.Route
  * preferred.
  */
 class RouteDatabase {
+  /**
+   * Don't grow unbounded, especially since the [SSLSocketFactory] instances held by [Route] may
+   * be quite large: 270 KiB according to a report on issue 5746.
+   */
+  private val maxSize = 128
+
   private val failedRoutes = mutableSetOf<Route>()
```

Comment:
May as well presize to maxSize and use direct constructor `LinkedHashSet(maxSize)`